tsfc/driver.py

```diff
@@ -241,7 +243,11 @@ def compile_integral(idata, fd, prefix, parameters):
     return kernel
 
 
-def prepare_coefficient(integral_type, coefficient, name):
+def prepare_coefficient(integral_type, coefficient, name, mode=None):
+    if mode is None:
+        mode = 'manual_loop'
+
+    assert mode in ['manual_loop', 'list_tensor']
```

Comment:
Is there a reason not to always use one over the other (and when should one make the choice?)